wezterm-gui/src/selection.rs

```diff
@@ -77,10 +78,12 @@ impl SelectionRange {
                     start: SelectionCoordinate {
                         x: 0,
                         y: logical.first_row,
+                        ..Default::default()
```

Comment:
Most of this PR is adding in this `..Default::default()` which makes it more difficult to see where the actual logic changes are, and makes the code a bit more verbose in general.

My suggestion is to add a constructor as a shortcut for the common case:

```rust
impl SelectionCoordinates {
   pub const fn x_y(x: usize, y: StableRowIndex) -> Self {
       Self {
            x,
            y,
            treat_zero_in_x_as_before_first_cell: false
        }
    }
}
```

then most of the changes turn into something a bit more compact like:

```rust
   start: SelectionCoordinate::x_y(0, logical.first_row),
```

Reply:
Done